### PR TITLE
Switch NUsight build to Vite + Rollup

### PR DESCRIPTION
**Pull request by JosephusPaye**
This PR migrates the NUsight build system from Webpack to Vite/Rollup. Webpack remains functional for Storybook. There's an experimental [Vite-powered builder for Storybook](https://storybook.js.org/blog/storybook-for-vite/) that we can possibly migrate to once it's stable.

All build related commands have been migrated and tested. `yarn test`, `yarn prod`, `yarn tscheck`, `yarn storybook`, etc. continue to work as expected.

It's ready for review/feedback.

### CODE REVIEW

File: nusight2/src/client/components/app/view.tsx [hunk 1]
```diff
@@ -20,11 +20,9 @@ export class AppView extends Component<{ nav: NavigationConfiguration }> {
             <div className={style.app__content}>
               <Suspense fallback={<div>Loading...</div>}>
                 <Routes>
-                  {...this.props.nav
```
Comment (JosephusPaye) [inline nusight2/src/client/components/app/view.tsx:23]: Spreading of children here generated an error.

File: nusight2/src/client/components/checkbox/view.tsx [hunk 2]
```diff
@@ -13,7 +12,7 @@ export interface CheckboxProps {
   onChange(event: ChangeEvent<HTMLInputElement>): void
 }
 
-export const Checkbox: StatelessComponent<CheckboxProps> = (props: CheckboxProps) => {
```
Comment (JosephusPaye) [inline nusight2/src/client/components/checkbox/view.tsx:16]: `StatelessComponent` is now `PureComponent`. We don't need the type here though.

File: nusight2/src/client/components/localisation/nugus_robot/view_model.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ import { GLTFLoader } from 'three/examples/jsm/loaders/GLTFLoader'
 
 import { LocalisationRobotModel } from '../darwin_robot/model'
 
-import url from './config/nugus.glb'
+import url from './config/nugus.glb?url'
```
Comment (JosephusPaye) [inline nusight2/src/client/components/localisation/nugus_robot/view_model.ts:13]: Vite compatible URL import

File: nusight2/src/client/nuclearnet/webworker_web_socket_client.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import Emitter from 'component-emitter'
 import { SocketOptions } from 'socket.io-client'
 
-import WebSocketWorker from './webworker_web_socket_client.worker'
+import WebSocketWorker from './webworker_web_socket_client.worker?worker'
```
Comment (JosephusPaye) [inline nusight2/src/client/nuclearnet/webworker_web_socket_client.ts:4]: Vite compatible worker import

File: nusight2/tsconfig.json [hunk 5]
```diff
@@ -22,7 +22,8 @@
     "typeRoots": [
       "./node_modules/@types",
       "./typings"
-    ]
+    ],
+    "rootDir": "./"
```
Comment (JosephusPaye) [inline nusight2/tsconfig.json:26]: Needed for the typescript rollup plugin when building the server

File: .github/workflows/nusight.yaml [hunk 6]
```diff
@@ -141,7 +141,7 @@ jobs:
 
       - name: Build NUsight
         working-directory: nusight2
-        run: yarn build:ci
+        run: yarn tscheck && yarn build:ci
```
Comment (JosephusPaye) [inline .github/workflows/nusight.yaml:144]: Build no longer typechecks the client, this adds it for CI

File: nusight2/index.html [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <title>NUsight2</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="shortcut icon" href="/favicon.ico" />
+  </head>
+  <body>
+    <div id="root"></div>
```
Comment (LachlanCourt) [inline nusight2/index.html:10]:
```suggestion
    <div id="root" />
```

File: nusight2/server.rollup.config.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+  if (args['virtual-robots']) {
+    nonRollupArgs.push('--virtual-robots')
+  }
+
+  if (args['address']) {
+    nonRollupArgs.push('--address', args['address'])
+  }
+
+  return {
+    run: Boolean(args['run']),
```
Comment (LachlanCourt) [inline nusight2/server.rollup.config.js:31]: What's the benefit of this over `!!args['run']` ? I've never seen it done like this before

**Comment by JosephusPaye** [comment]
Superseded by the minor patch in #1073

**Pull request closed without merging**